test/runIntegrationTests.ts

```diff
@@ -0,0 +1,74 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as cp from 'child_process';
+import * as path from 'path';
+import { downloadAndUnzipVSCode, resolveCliArgsFromVSCodeExecutablePath, runTests } from '@vscode/test-electron';
+import { execChildProcess } from '../src/common';
+
+function getSln(workspacePath: string): string | undefined {
+    if (workspacePath.endsWith('slnWithGenerator')) {
+        return 'slnWithGenerator.sln';
+    }
+    return undefined;
+}
+
+async function main() {
+    try {
+        const vscodeExecutablePath = await downloadAndUnzipVSCode('stable');
```

Comment:
we should share this code with the O# version since this should be basically the same (even with jest).  We can probably keep only the copy in the 'test' folder.  We will need to switch certain variables based on which test is running (e.g. 'extensionTestsPath')